### PR TITLE
Launchpad recall

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Adds in a recall timer to the bluespace launchpad computer.

## Why It's Good For The Game

This means that the computer is now useful if you are a solo antagonist and want to do a heist on part of the station, but it has an extreme amount of risk:
- If you miss the teleport, you are stuck
- You are forced to wait for the timer you set, so if you need to teleport out FAST then you will have to wait.

## Testing Photographs and Procedure

https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/16a59a29-2348-42f3-94d3-0445615b3518

(I made it so that the UI auto-updates when the number changes)

## Changelog
:cl:
balance: Launchad computer can now set a timer to automatically recall after a set amount of time, up to 5 minutes.
/:cl:

### CODE REVIEW

**Comment by Harranhall** [comment]
This is so perfect.

**Comment by Admiral-Gigaton** [comment]
this is awesome, next time im in an incursion im slapping this shit on a shuttle and sucking up all of ai sat, armory, vault, and sending a bomb team to plant explosives on ian

**Comment by Harranhall** [comment]
> this is awesome, next time im in an incursion im slapping this shit on a shuttle and sucking up all of ai sat, armory, vault, and sending a bomb team to plant explosives on ian

You'll have to teleport yourself in with a duffel bag big guy, the pull function only works on entities now, bot items.
Theres also a bluespace ripple visible on teleport now as well.

File: code/game/machinery/computer/launchpad_control.dm [hunk 1]
```diff
@@ -256,8 +167,110 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/computer/launchpad)
...
 		if("launch")
 			teleport(usr, current_pad, TRUE)
+			if (recall_time)
+				addtimer(CALLBACK(src, PROC_REF(teleport), usr, current_pad, FALSE), recall_time * 10)
 			. = TRUE
 
 		if("pull")
 			teleport(usr, current_pad, FALSE)
+			if (recall_time)
+				addtimer(CALLBACK(src, PROC_REF(teleport), usr, current_pad, TRUE), recall_time * 10)
```
Comment (PowerfulBacon) [inline code/game/machinery/computer/launchpad_control.dm]: Should pull really send the object back after the timer expires?

Comment (Liver-1) [reply code/game/machinery/computer/launchpad_control.dm]: I can't think of a reason why that would be a wanted or expected effect.

Comment (PowerfulBacon) [inline code/game/machinery/computer/launchpad_control.dm]:
```suggestion
```

File: code/game/machinery/computer/launchpad_control.dm [hunk 2]
```diff
@@ -256,8 +167,108 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/computer/launchpad)
...
 
 		if("pull")
 			teleport(usr, current_pad, FALSE)
 			. = TRUE
+
+/obj/machinery/computer/launchpad/Initialize(mapload)
+	launchpads = list()
+	. = ..()
+	AddComponent(/datum/component/usb_port, list(
+		/obj/item/circuit_component/bluespace_launchpad,
```
Comment (itsmeow) [inline code/game/machinery/computer/launchpad_control.dm:182]: why move all this code fro no reason?

Also, the recall_time should be included in the USB interface.

Comment (PowerfulBacon) [reply code/game/machinery/computer/launchpad_control.dm:182]: Because it was in the middle of the launchpad code for some reason

**Review by Rukofamicom — APPROVED** [review]
👍

**Pull request merged**